Server/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -404,56 +405,33 @@ int CLuaPedDefs::IsPedFrozen(lua_State* luaVM)
-    int       iBlend = 250;
-    bool      bLoop, bUpdatePosition, bInterruptable, bFreezeLastFrame;
-    bool      bDummy;
-    bool      bTaskToBeRestoredOnAnimEnd;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pPed);
-    if (argStream.NextIsBool())
-        argStream.ReadBool(bDummy);            // Wiki used setPedAnimation(source,false) as an example
-    else if (argStream.NextIsNil())
-        argStream.m_iIndex++;            // Wiki docs said blockName could be nil
-    else
-        argStream.ReadString(strBlockName, "");
-    argStream.ReadString(strAnimName, "");
-    if (argStream.NextCouldBeNumber())            // Freeroam skips the time arg sometimes
-        argStream.ReadNumber(iTime, -1);
-    else
-        iTime = -1;
-    argStream.ReadBool(bLoop, true);
-    argStream.ReadBool(bUpdatePosition, true);
-    argStream.ReadBool(bInterruptable, true);
-    argStream.ReadBool(bFreezeLastFrame, true);
-    argStream.ReadNumber(iBlend, 250);
-    argStream.ReadBool(bTaskToBeRestoredOnAnimEnd, false);
+    std::string animBlockName = "";
+    std::string animationName = "";
 
-    if (!argStream.HasErrors())
+    if (blockName.has_value())
     {
-        const char *szBlock, *szAnim;
-        szBlock = strBlockName.empty() ? NULL : strBlockName.c_str();
-        szAnim = strAnimName.empty() ? NULL : strAnimName.c_str();
+        if (std::holds_alternative<std::string>(blockName.value()))
+            animBlockName = std::get<std::string>(blockName.value());
+        else if (std::holds_alternative<bool>(blockName.value()))
+            if (std::get<bool>(blockName.value()))
+                throw LuaFunctionError("Anim block name cannot be true. Possible values: nil, false, string.");
+    }
 
-        if (CStaticFunctionDefinitions::SetPedAnimation(pPed, szBlock, szAnim, iTime, iBlend, bLoop, bUpdatePosition, bInterruptable, bFreezeLastFrame,
-                                                        bTaskToBeRestoredOnAnimEnd))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
+    if (animName.has_value())
+    {
+        if (std::holds_alternative<std::string>(animName.value()))
+            animationName = std::get<std::string>(animName.value());
+        else if (std::holds_alternative<bool>(animName.value()))
+            if (std::get<bool>(animName.value()))
+                throw LuaFunctionError("Animation name cannot be true. Possible values: nil, false, string.");
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return CStaticFunctionDefinitions::SetPedAnimation(pPed, animBlockName.empty() ? NULL : animBlockName.c_str(), animationName.empty() ? NULL : animationName.c_str(), time.value_or(-1), blendTime.value_or(250), loop.value_or(true), updatePosition.value_or(true), interruptable.value_or(true), freezeLastFrame.value_or(true), restoreTask.value_or(false));
```

Comment:
Is the ternary operator really needed here? If a string is empty you can just pass it as is.